src/main/java/com/luv2code/ecommerce/controller/CheckoutController.java

```diff
@@ -35,15 +38,18 @@ public PurchaseResponse placeOrder(@RequestBody Purchase purchase) {
         return purchaseResponse;
     }
 
+    // comments
     @PostMapping("/payment-intent")
     public ResponseEntity<String> createPaymentIntent(@RequestBody PaymentInfo paymentInfo) throws StripeException {
```

Comment:
Tu dodaj validaciju. Nisam siguran kako napamet, ali ako se ne varam potrebno je kao argument napisati:
`@RequestBody @Validate PaymentInfo paymentInfo`.

To ti omogucava da u `PaymentInfo` (kako tu, tako i u ostalim klasma koje koristis kao body), klasi dodat validacijske anotacije i automatski se vrati greska ako je koristnik nesto krivo poslao. NAJBITNIJE PRAVILO backend developmenta je, NIKADA ne vjeruj korisniku (cak i ako si taj korisnik ti), Uvijek je potrebno provjeriti podatke,